compiler/passes/initializerRules.cpp

```diff
@@ -2318,3 +2328,61 @@ FnSymbol* buildClassAllocator(FnSymbol* initMethod) {
 
   return fn;
 }
+
+static void buildTypeFunction(FnSymbol* fn) {
+  Symbol* _this = fn->_this;
+  AggregateType* at = toAggregateType(_this->type);
+
+  SET_LINENO(at);
```

Comment:
It is sometimes tricky to decide what to use for a line number.  I wonder if it would be better
to tie the type-function to the initializer that generates it rather than to the definition of the
type (which could be in a different file etc)?